src/node/coinstats.h

```diff
@@ -69,12 +66,10 @@ struct CCoinsStats {
     CAmount total_unspendables_scripts{0};
     //! Total cumulative amount of coins lost due to unclaimed miner rewards up to and including this block
     CAmount total_unspendables_unclaimed_rewards{0};
-
-    CCoinsStats(CoinStatsHashType hash_type) : m_hash_type(hash_type) {}
 };
 
 //! Calculate statistics about the unspent transaction output set
-bool GetUTXOStats(CCoinsView* view, node::BlockManager& blockman, CCoinsStats& stats, const std::function<void()>& interruption_point = {}, const CBlockIndex* pindex = nullptr);
+bool GetUTXOStats(CCoinsView* view, node::BlockManager& blockman, CCoinsStats& stats, CoinStatsHashType hash_type, const std::function<void()>& interruption_point = {}, const CBlockIndex* pindex = nullptr);
```

Comment:
I think this is completely changed when we introduce `ComputeUTXOStatsWithHasher(UTXOHasher& hasher, ...)` with `UTXOHasher&` as the first parameter. Does that look alright?

If it still looks unclear, I'm happy to add Doxygen comments so that's it's more clear!

Reply:
I've experimented with this on top of the latest push 90358d4b7402bfd4cd9b79ebf828e40f4eabd849, but the template parameter would be based on a variable and that's not allowed in C++ (at least in clang).